EIPS/eip-1102.md

```diff
@@ -69,18 +61,15 @@ If a user rejects web3 access on an untrusted site, the site itself MUST NOT be
 The following example demonstrates one possible implementation of this strategy in a browser-based DOM environment. Note that web3 environments on other platforms would most likely use platform-specific native messaging protocols, not `postMessage`.
 
 ```js
-if (typeof web3 !== 'undefined') {
-    // web3 API defined, start dapp
-} else {
-    window.addEventListener('message', function (event) {
-        if (!event.data || !event.data.type) { return; }
-        if (event.data.type === 'WEB3_API_SUCCESS') {
-            // web3 API defined, start dapp
-        }
-    });
-    // request web3 API
-    window.postMessage({ type: 'WEB3_API_REQUEST' });
-}
+// listen for web3 provider
+window.addEventListener('message', function(event) {
+  if (event.data && event.data.type === 'REQUEST_ETHEREUM_PROVIDER') {
+    // supplied provider, start dapp
+    startDapp(event.data.ethereum);
```

Comment:
Instead of calling a finctional `startDapp` function, an we add a comment similar to the following to give insight into real-world use?

```
// provider API defined, use with web3.js or similar
```

Reply:
Same comment, `web3.js` may not be around in the future (from what I have seen it is already losing favor to alternatives).  We should try to use language that will stand the test of time (`dapp` is better, though I think for an example I would do something like:
```typescript
if (...) {
    const networkVersion = await event.data.ethereum.send('net_version', [])
    console.log(networkVersion)
}
```

Reply:
I like the idea of showing example usage as opposed to setting the provider on a specific library.